### PR TITLE
Voice rooms prototype

### PR DESCRIPTION
**Pull request by robintown**
See https://github.com/matrix-org/matrix-react-sdk/pull/8084 for the full details.

Notes: Support call room type from MSC3417

---
This PR currently has no changelog labels, so will not be included in changelogs.

Add one of: `T-Deprecation`, `T-Enhancement`, `T-Defect`, `T-Task` to indicate what type of change this is plus `X-Breaking-Change` if it's a breaking change.

### CODE REVIEW

File: src/models/room.ts [hunk 1]
```diff
@@ -2344,6 +2344,14 @@ export class Room extends TypedEventEmitter<EmittedEvents, RoomEventHandlerMap>
         return this.getType() === RoomType.Space;
     }
 
+    /**
+     * Returns whether the room is a call-room as defined by MSC3417.
+     * @returns {boolean} true if the room's type is RoomType.Call
```
Comment (SimonBrandner) [inline src/models/room.ts:2349]: I don't know why but this feels better here?

```suggestion
     * @returns {boolean} true if the room's type is m.call
```

**Review by SimonBrandner — APPROVED** [review]
LGTM - thank you!

File: src/models/room.ts (hunk 1 above)
Comment (robintown) [reply src/models/room.ts:2349]: The room type isn't `m.call` at the moment, though - it's `org.matrix.msc3417.call`. Also, the `isSpaceRoom` docs above say the same thing.

Comment (SimonBrandner) [reply src/models/room.ts:2441]: Ah, ok, _resolved_

File: src/@types/event.ts [hunk 2]
```diff
@@ -119,6 +119,7 @@ export const RoomCreateTypeField = "type";
 
 export enum RoomType {
     Space = "m.space",
+    Call = "org.matrix.msc3417.call",
```
Comment (turt2live) [inline src/@types/event.ts:122]: I'd suggest this be called `UnstableCall` so we can introduce the stable version more easily down the line. Alternatively, `UnstableValue` usage outside of this enum (like we do for event types).

**Review by turt2live — COMMENTED** [review]
generally looks fine on this side - thanks!

**Review by turt2live — APPROVED** [review]
this side lgtm

**Pull request merged**